common/types.go

```diff
@@ -126,9 +126,11 @@ type Database struct {
 type Pipeline struct {
 	Tags   map[string]interface{} `yaml:"tags,omitempty"`
 	Source struct {
-		Provider string `yaml:"provider,omitempty"`
-		Repo     string `yaml:"repo,omitempty"`
-		Branch   string `yaml:"branch,omitempty"`
+		Provider        string `yaml:"provider,omitempty"`
+		Repo            string `yaml:"repo,omitempty"`
+		Branch          string `yaml:"branch,omitempty"`
+		SourceBucket    string `yaml:"sourcebucket,omitempty"`
```

Comment:
I'd prefer to follow the same pattern with CodeCommit/GitHub where the `repo` field is overloaded based on provider.  So could we remove SourceBucket and SourceObjectKey and have Repo be something like `<bucket_name>/<object_key>` and then internally we split that apart before passing into the CFN?